### DISCUSSION TITLE
Adding covariates into the model

### DISCUSSION THREAD

**Discussion by VishnuEriyat** [LAPKB/Pmetrics#207] [Q&A] [answered]
I was trying to add the covariates into the model, while trying to build a model for rifampicin. When I add the covariate data columns into the data csv file and start the run, the execution is halted at the early part itself. The error message snip is attached here. The following are what I tried to resolve this issue, none worked till now,
1. I went through the covariate data looking for any spaces, missing values other characters etc
2. Checked the model file #Cov block for any spelling mistake, order change etc
3. I tried adding covariate columns one by one, anyone variable I added it failed
4. Copypasted the entire data into another excel saved it with a new file name and tried
5. Tried the model file with  data csv file prepared by a friend(similar data with the same covariates in the same order)
6. Just put a covariate name and added some numerical value there and tried.
None of these seems to be working
[Covariate adding error.pdf](https://github.com/LAPKB/Pmetrics/files/14404703/Covariate.adding.error.pdf)

**Comment by mhovd** [comment]
Could you try the same with one subject only in the datafile, where you are sure the covariates are specified correctly? I.e. no characters and no missing values at time 0.

Are all the covariates in your datafile included in the `#COV` block of the generated model?

**Reply by VishnuEriyat** [reply, marked as answer]
**The curious case of #ID**
Ya, tried that and it didn't workout, I had even tried typing in manually and running few subject data, then also the execution was halted. Finally I was just comparing my previous data file and current one, then I found out that the ID column in the upper corner was 'ID' instead of '#ID'. Even with this the usual run without covariate was going on fine. When I added the '#' then the run is going on smooth with covariates added.

**Reply by mhovd** [reply]
Yes, the `#ID` column requires the `#` to be there. In the future, if you create your data with `data = PM_data$new(...)`, you can write Pmetrics compatible data to file with `data$write("filename.csv")`.

**Reply by VishnuEriyat** [reply]
Sure, Will try it out,Thanks

**Comment by mnneely** [comment]
@VishnuEriyat , what version of Pmetrics are you using? Prior to version 2, the “#” was indeed required before ID as it was a marker for the header row. We eliminated that in version 2, and in fact, at minimum you now only need 4 columns in your data: ID, TIME, DOSE, OUT.  Names of these columns can’t be altered, but the “#” is not required. As, @mhovd said though, you need to use the `PM_data$new(filename)` construct to properly process such a limited .csv file and make it available in R as a `PM_data` object to use in fitting or simulating.

**Reply by VishnuEriyat** [reply]
Hi, I am using Pmetrics, version 2.1.1. , is it that I have 2 different versions installed in the system, but 2.1.1 is what I get when I call library(Pmetrics)
